File: src/main/java/com/itsci/mju/maebanjumpen/MaebanjumpenApplication.java
package com.itsci.mju.maebanjumpen;

import com.fasterxml.jackson.databind.SerializationFeature; // อาจจะมีหรือไม่มีก็ได้
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration; // อาจจะมีหรือไม่มีก็ได้
import org.springframework.scheduling.annotation.EnableScheduling;

@SpringBootApplication
@EnableScheduling
public class MaebanjumpenApplication {

	public static void main(String[] args) {
		SpringApplication.run(MaebanjumpenApplication.class, args);
	}

	@Bean
	public Jackson2ObjectMapperBuilderCustomizer jsonCustomizer() {
		return builder -> {
			builder.modules(new JavaTimeModule()); // บรรทัดนี้สำคัญมาก
			// builder.featuresToDisable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS); // บรรทัดนี้จะไม่มีผลกับปัญหาปัจจุบัน
		};
	}
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/MaebanjumpenApplication.kt
package com.itsci.mju.maebanjumpen

import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.scheduling.annotation.EnableScheduling

@SpringBootApplication
@EnableScheduling
class MaebanjumpenApplication {

    @Bean
    fun jsonCustomizer(): Jackson2ObjectMapperBuilderCustomizer {
        return Jackson2ObjectMapperBuilderCustomizer { builder ->
            builder.modules(JavaTimeModule())
        }
    }
}

fun main(args: Array<String>) {
    runApplication<MaebanjumpenApplication>(*args)
}

